src/mbgl/tile/vector_tile.cpp

```diff
@@ -15,9 +15,17 @@ class VectorTileLayer;
 
 using packed_iter_type = protozero::iterator_range<protozero::pbf_reader::const_uint32_iterator>;
 
+struct VectorTileLayerData {
```

Comment:
@anandthakker and @jfirebaugh - awesome thank you for the confirmation. I'm grateful for the confirmation and will now proceed with my PR revising this code and keep all the explicit ownership relationships fully intact. The context I should have mentioned before is that I hope my PR will help performance so I was looking closely at the nearby code for any optimization potential, which is what lead me to the question of whether the shared pointers were absolutely needed. Great to know they are.